test/common/stats/thread_local_store_test.cc

```diff
@@ -1186,5 +1195,196 @@ TEST_F(HistogramTest, ParentHistogramBucketSummary) {
             parent_histogram->bucketSummary());
 }
 
+class ClusterShutdownCleanupStarvationTest : public ThreadLocalStoreNoMocksTestBase {
+public:
+  static constexpr uint32_t NumThreads = 1; // Test seems to go fastest with 1 worker thread.
+  static constexpr uint32_t NumScopes = 2000;
+  static constexpr uint32_t NumIters = 35;
+
+  // Helper class to block on a number of multi-threaded operations occurring.
+  class BlockingScope {
```

Comment:
nit: 'Scope' is overloaded in stats and initially I was confused that this was about something stats related. Can you rename to something else? BlockingBarrier? Something else?

Reply:
agreed; will do.